### DISCUSSION TITLE
[Feature Request] Toggle "Ignore" mode on and off with a shortcut

### DISCUSSION THREAD

**Discussion by informatik01** [rxhanson/RectanglePro-Community#477] [Ideas]
It would be really nice to be able to _toggle_ Rectangle on or off with a _shortcut_, ideally configurable shortcut to avoid conflicts with other shortcuts. The idea is that it would be possible to assign some "exotic" key combination for switching Rectangle on and off not to conflict with the existing shortcuts in the current app.

It would be useful when using IDEs (in my case these are JetBrains products) and code editors (e.g. Visual Studio Code) which have quite a few similar shortcuts as Rectangle.

At the moment there is an option to "Ignore " which has to be manually triggered in case of a need, but it would be really convenient to ignore some app most of the time and only when moving app windows is needed to quickly switch Rectangle on with a shortcut, perform the desired move (also with a shortcut) and switch Rectangle off for the current app again.

**Comment by rxhanson** [comment]
Thanks for the feedback. I'll add this one to my backlog.

Just so I understand better, is there a reason that you prefer the mental overhead of overlapping shortcuts over remapping the default shortcuts in Rectangle or other apps? Perhaps there is a better solution to this problem.

**Comment by informatik01** [comment]
Well, the initial reason was that my set of shortcuts in different IDEs (specifically IntelliJ IDEA), code editors (specifically VS Code) and database clients (specifically DataGrip, DBeaver, TablePlus) conflicted with some of the Rectangle shortcuts. And in all these apps there were _different_ conflicts.

So _by default_ I have Rectangle running in the background (on macOS). And only for the above listed apps I have Rectangle set to "Ignore <ap_name>". When I need Rectangle to perform some window manipulating actions I temporarily switch it on, perform an action and switch it back to "Ignore" mode.

In general I like the predefined Rectangle shortcuts, very convenient and I already got used to them. It's just that IDEs and other programming tools usually have a large set of different shortcuts (which in turn you _may_ also customize additionally to avoid conflicts with OS / other app shortcuts) which also differ from app to app. So the ability to toggle Rectangle's "Ignore" mode **_on demand_** naturally seemed really convenient to me.

In other words instead of redefining _multiple_ shortcuts, would be convenient to have "one shortcut to rule them all". Of course **IF** it won't involve much effort on your side / won't cause other issues / easy to implement etc.

**Reply by rxhanson** [reply]
Thanks for the additional info, I appreciate it. Your logic makes sense; I was hoping maybe I could think of a solution that I like better than both of these options. There's always friction with remapping shortcuts, and having to mentally keep track of state is also not desirable, but at the moment I can't think of anything better.

**Comment by chrisortman** [comment]
I was trying to figure out how to make something like this work as well. Instead of ignoring certain apps, I'd like a command to enable rectangle keyboard shortcuts and turn them off. Almost like it is an editor mode. 

For example (in Bear ctrl+command is used for table manipulation) but I also like using ctrl+command + arrow for window management. 
So I would press something like ctrl+command+w and that would put rectangle into keypress mode where I might hit arrow keys or use the U,I,J,K to position a window then hit esc or enter and leave the mode.

**Comment by bhparsons** [comment]
Would love an update on this thread. The clicks required to turn things on and off are the biggest downside of rectangle pro for me right now. Would love to have a simple shortcut to turn the other commands on and off.

**Comment by rxhanson** [comment]
I'll bump this up in priority and see if I can get to it soon.
